elb.go

```diff
@@ -53,13 +54,17 @@ func (tg *AWSTargetGroup) GetTargets(ctx context.Context) ([]*Target, error) {
 	}
 
 	targets := make([]*Target, 0)
-	for _, targetHealthDecription := range result.TargetHealthDescriptions {
-		if tg.cfg.AvailabilityZone == "" ||
-			*targetHealthDecription.Target.AvailabilityZone == tg.cfg.AvailabilityZone {
-			targets = append(targets, &Target{
-				IP:   *targetHealthDecription.Target.Id,
-				Port: int(*targetHealthDecription.Target.Port),
-			})
+	for _, healthDesc := range result.TargetHealthDescriptions {
+		if tg.cfg.AvailabilityZone == "" || *healthDesc.Target.AvailabilityZone == tg.cfg.AvailabilityZone {
+			state := *healthDesc.TargetHealth.State
+			if state == elbv2.TargetHealthStateEnumInitial || state == elbv2.TargetHealthStateEnumHealthy {
```

Comment:
nit: I'd switch on the state here and then you could have a single case with these all on the same line (it'll be easier to read)